### PR TITLE
Serialize oep data

### PR DESCRIPTION
**Pull request by jh-RLI**
- [ ] 

- [ ] 

- [ ] 

- [ ]

### CODE REVIEW

File: app_settings.py [hunk 1]
```diff
@@ -1 +1,20 @@
...
+
+wam_config = settings.config['DATABASES']['OEP']
+
+############################################SQLAlchemy ENGINE#####+#####################################################
+# db connection string for sqlalchemy-engine
+DB_URL = '{ENGINE}://{USER}:{PASSWORD}@{HOST}:{PORT}/{NAME}'.format(**wam_config)
+
+engine = create_engine(DB_URL)
+sqlahelper.add_engine(engine, 'oep_engine')
+########################################################################################################################
```
Comment (jh-RLI) [inline app_settings.py:20]: This sets up an engine that is used by sqlalchemy f.e. to query the OEP database.

File: oep_models.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        location = relationship(classes["Location"], backref="series")
+        variable = relationship(classes["Variable"], backref="series")
+
+    classes["Series"] = Series
+
+    return classes
+
+
+# contains open_FRED related tables as sqla class
+classes = mapped_classes(metadata)
```
Comment (jh-RLI) [inline oep_models.py:137]: This skript creates the tables as sqlalchemy modle. This actually takes some time every time the server is started (about 3 min). Got the skript from [this repo/file](https://github.com/open-fred/cli/blob/master/openFRED.py).

File: serial_views.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import WAM_APP_FRED.oep_models as oep_modles
+# from WAM_APP_FRED.db_sqla import *
+from shapely.wkb import loads as loadswkb
+from geojson import Feature, FeatureCollection, dumps
+from geoalchemy2 import functions
+import sqlahelper as sah
+from datetime import datetime
+
+# def serializer():
+#     """
```
Comment (jh-RLI) [inline serial_views.py:14]: This is a simple integration of my serializer should be easy to understand . Will delete this soon.

File: serial_views.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+#     for record in session.query(oep_modles.classes['Location']).limit(100):
+#         geometry = loads(str(record.point), True)
+#         propertys = record.id
+#         feature = Feature(id=record.id, geometry=geometry)
+#
+#         features.append(feature)
+#
+#     return dumps(FeatureCollection(features))
+
+class Serializer():
```
Comment (jh-RLI) [inline serial_views.py:36]: Created the serializer as class so one can better understand what kind of data is in a GeoJSON that is returned by a class method.

File: serial_views.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+            geometry = loadswkb(str(record.Series.location.point), True)
+            feature = Feature(id=record.Series.id, geometry=geometry)
+            features.append(feature)
+
+        return HttpResponse(dumps(FeatureCollection(features)), content_type="application/json")
+
+
+
+    def wseries_property_view(self):
```
Comment (jh-RLI) [inline serial_views.py:72]: created different methods that return the geometry or the propertys for each database record in order to reduce the amount of data in a single GeoJSON.

File: serial_views.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        #         for record in Serializer.session.query(oep_modles.classes['Series'],
+        #                                                oep_modles.classes['Timespan'],
+        #                                                oep_modles.classes['Variable'])\
+        #                                                 .join(oep_modles.classes['Timespan']) \
+        #                                                 .join(oep_modles.classes['Variable']).limit(1000):
+        #
+        #
+        #             return HttpResponse(geojsondata, content_type="application/json")
+
+    def pp_list_geometry_view(self):
```
Comment (jh-RLI) [inline serial_views.py:132]: already added the example function`s to keep in mind what needs to be done.

File: settings.py [hunk 7]
```diff
@@ -1,2 +1,32 @@
...
+                'minZoom': 9,
+                })
+              ],
+    # Germany center (center = long/2 and lat/2)
+    'DEFAULT_CENTER': (53.4554, 9.6211),
+    'DEFAULT_ZOOM': 16,
+    'MAX_ZOOM': 50,
+    'RESET_VIEW': False,
+    'NO_GLOBALS': False,
+}
```
Comment (jh-RLI) [inline settings.py:32]: This needs to be adjusted, copied from other repository. Not sure if it is used at the moment.

File: urls.py [hunk 8]
```diff
@@ -1,8 +1,12 @@
...
+from WAM_APP_FRED.serial_views import Serializer
 app_name = 'WAM_APP_FRED'
 
 urlpatterns = [
-    path('', views.fred_map, name='index')
+    # path('', views.fred_map, name='index')
+    path('', views.webgui_test, name='index'),
 
+    # just point geometrys as geojson featureCollection
+    path('Locations.data/', Serializer.wseries_geometry_view, name='Locations.data')
```
Comment (jh-RLI) [inline urls.py:11]: This is the URL that provides the geojson in the template.

File: db_sqla.py [hunk 9]
```diff
@@ -1,203 +1,43 @@
-import os
```
Comment (Bachibouzouk) [inline db_sqla.py:1]: This file should be deleted as per our discussion

File: oep_models.py [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+############################################TABLE DEFINITION############################################################
+
+# included function from github: https://github.com/open-fred/cli/blob/master/openFRED.py
+# copied function to aviode dependencies
+def mapped_classes(metadata):
+    """ Returns classes mapped to the openFRED database via SQLAlchemy.
+
+    The classes are dynamically created and stored in a dictionary keyed by
+    class names. The dictionary also contains the special entry `__Base__`,
+    which an SQLAlchemy `declarative_base` instance used as the base class from
```
Comment (Bachibouzouk) [inline oep_models.py:43]:
```suggestion
    which is an SQLAlchemy `declarative_base` instance, used as the base class from
```

File: serial_views.py (hunk 4 above)
Comment (Bachibouzouk) [reply serial_views.py:36]: I am not sure what you mean here

File: serial_views.py [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+            # timespan_collection = {"Start": record.Series.timespan.start.strftime('%b %d %Y %I:%M%p'),
+            #                        "Stop": record.Series.timespan.stop.strftime('%b %d %Y %I:%M%p'),
+            #                        "Resolution": record.Series.timespan.resolution}
+
+            netcdf = record.Series.variable.netcdf_attributes
+            variables_collection = {"Name": record.Series.variable.name}
+            variables_collection["NetCDF"] = netcdf
+
+
+            propertys = {"SeriesID": record.Series.id, "values": record.Series.values, "height": record.Series.height}
```
Comment (Bachibouzouk) [inline serial_views.py:103]: plural of `property` is `properties`

File: serial_views.py (hunk 6 above)
Comment (Bachibouzouk) [reply serial_views.py:132]: "the example function" --> "empty functions"

File: settings.py (hunk 7 above)
Comment (Bachibouzouk) [reply settings.py:32]: Which repository?

File: wam_environment.yml [hunk 12]
```diff
@@ -0,0 +1,16 @@
+#this should extend the django-wam environment
```
Comment (Bachibouzouk) [inline wam_environment.yml]: This file can be deleted as it will be in #14

**Review by Bachibouzouk — APPROVED** [review]
I made a few suggestions that you are free to implement or not. What is more important is the following

Before merging : 
- [x] merge `develop` into this branch (locally)
- [x] delete the db_sqla.py and wam_environment.yml files
- [x] update the CHANGELOG file
- [x] push your changes
- [x] click the merge button ;) 

After merging:
- [ ] update other PRs by merging the latest version of `develop`

**Pull request merged**